com.github.emmanueltouzery.projectpad.json

```diff
@@ -0,0 +1,99 @@
+{
+  "app-id": "com.github.emmanueltouzery.projectpad",
+  "runtime": "org.freedesktop.Platform",
+  "runtime-version": "19.08",
+  "sdk": "org.freedesktop.Sdk",
+  "sdk-extensions": ["org.freedesktop.Sdk.Extension.rust-stable"],
+  "command": "projectpad",
+  "finish-args": [
+    "--share=ipc",
+    "--socket=fallback-x11",
+    "--socket=wayland",
+    "--device=dri",
+    "--filesystem=home",
```

Comment:
Why does need access to home?

Reply:
* it stores a database in the home folder which can then be used by other applications (concretely, the sister ppcli tool)
* you can add files (for instance keys) in the application's database
* you can associate pictures for projects, meaning pick files on the local hard drive

Reply:
I think for 2 and 3 it would be better to use GtkFileChooserNative which uses portal for all files operations. This way there's no need to specify filesystem permissions.

As for 1, flatpak grants write access to `$XDG_DATA_HOME` by default.

Reply:
I've looked at $XDG_DATA_HOME: it's `~/.var/app/com.github.emmanueltouzery.projectpad/data` when running in flatpak, but `~/.local/share/` when running native. I have tried many combinations (mostly huge hacks) so that I could keep supporting running the app as native as well as in flatpak, while having my external app (which is always native) always finding that database, and I failed. The problem is that the app which is not running in flatpak must guess the location used by flatpak, keeping in mind the other may not actually be using flatpak. It just complicated everything for me, I think my use-case doesn't fit in this case. Having a consistent and standard folder (and easily findable also from outside of flatpak) where data is stored is important in this case for me. So I'd just leave the home permission.

Reply:
I understand that it's pretty much `if $XDG_DATA_HOME then $XDG_DATA_HOME else ~/.local/share`. @bilelmoussaoui is it how GNOME apps usually handle it?

Reply:
My problem is that I split my app in two in part due to flatpak. In reality, it's impossible to do all that I want to do with this app within flatpak. So I made two apps: this one, which has a GUI and doesn't need to run on the host, and a CLI statically linked companion app that _must_ run on the host. And they share the database. And I'd like to support the case of the flatpak app being run on the metal too. That means that the CLI app must look in ~/.var/com.github... where the flatpak app saves, as well as ~/. local/share. And if a user wants to contribute to the app and run it on the host, I'd also like for the db to be found. It gets really messy. It is possible, but I got far enough that I thought that this system was not designed for that.

Reply:
Maybe if there was a not-too-hacky way for the app on the host to check whether flatpak is installed, and the com.github... app is installed, and what is the home sir of that app on the host.. but I doubt there is such an API or that you're supposed to do that

Reply:
ok, the current version right now doesn't work well. I used to store in $HOME/.projectpad and that worked well from outside flatpak and from the CLI app to. now I changed it to $XDG_DATA_HOME and that broke it, because it's different in the flatpak and out of the flatpak -- even if the flatpak has home permissions. I'll think about it a little more. The PR as it is is not OK.

Reply:
ok i think i have a work, if hackish, solution, without home privileges. i think it'll work out, i'm currently researching how to specify file extension filters in file dialogs through the portals. i'll write here when this is done (and presumably push a version of the json that does not request home privileges)

Reply:
ok, I finally made it work. I couldn't make file extension filtering, but I added a check for wrong file extensions in the app. The latest manifest doesn't ask for the home permissions anymore.